src/App.tsx

```diff
@@ -15,71 +16,108 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  name: string;
+  selectedItem: boolean;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    name: 'Jam',
+    selectedItem: true,
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  controlEvent = (event: React.MouseEvent<HTMLButtonElement>) => {
+    const retouch = event.currentTarget.innerText === '+';
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    this.clearAll();
+    if (retouch) {
+      const tdActive = event.currentTarget.parentElement;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      event.currentTarget.classList.add('is-info');
+      event.currentTarget.setAttribute('data-cy', 'RemoveButton');
+      tdActive?.parentElement?.classList.add('has-background-success-light');
+      // eslint-disable-next-line no-param-reassign
+      event.currentTarget.innerText = '-';
+      this.setState({ name: tdActive?.nextSibling?.textContent || '' });
+      this.setState({ selectedItem: true });
+    }
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  clearAll = () => {
+    const button = document.querySelectorAll('.button');
+    const tr = document.querySelectorAll('tr');
```

Comment:
Everything is much simpler here.
You don't need all those difficult calculations like in js. Refactor all your functions, please 
```
clearState = () => {
  this.setState({ selectedGood: '' });
};